src/core/ext/transport/inproc/inproc_transport.cc

```diff
@@ -765,7 +767,7 @@ void op_state_machine_locked(inproc_stream* s, grpc_error* error) {
       // any outstanding send_message op
       s->send_message_op->payload->send_message.send_message.reset();
       complete_if_batch_end_locked(
-          s, new_err, s->send_message_op,
+          s, GRPC_ERROR_EOS, s->send_message_op,
```

Comment:
Interesting that this case was never using stream_write_closed ; did it just behave differently because of the lack of flow control, some other reason, or just undertesting, I wonder.

Reply:
Yes, when I fixed it for chttp2, I hadn't fixed it for inproc. This PR fixes it for inproc too

Reply:
Does this need to be fixed in cronet too?

Reply:
Unfortunately, I did not finding testing for cronet in gRPC Core. Created https://github.com/grpc/grpc/issues/22693 for cronet